### PR TITLE
Feature/worksheet content

### PR DESCRIPTION
**Pull request by percyliang**
Major cleanup of worksheet content.
    
    - Get rid of state, use props when possible.
    - Remove indirection (item.state => item).
    - Remove cruft in WorksheetContent regarding reverse engineering alignments.
    - Better error handling when worksheet fails (still load things).
    - Remove checkboxes.

@pbhatnagar3 please review

### CODE REVIEW

File: codalab/apps/web/static/js/worksheet/worksheet_items.js [hunk 1]
```diff
@@ -82,11 +80,12 @@ var WorksheetItemList = React.createClass({
         this.props.updateWorksheetFocusIndex(index);  // Notify parent of selection (so we can show the right thing on the side panel)
         if (index != -1 && subIndex != null) {
           // Change subindex (for tables)
-          var item = this.state.worksheet.items[index].state;
-          if (item.mode == 'table') {
-            if (subIndex == 'end') {
-              subIndex = item.bundle_info.length - 1;  // Last row of table
-            }
+          var item = info.items[index];
+          console.log('EEEEEEEEEEE', item);
```
Comment (pbhatnagar3) [inline codalab/apps/web/static/js/worksheet/worksheet_items.js]: remove

**Comment by pbhatnagar3** [comment]
Changes looks good. Since major chunk of the worksheet content is now gone, I am not entirely certain if the updating of the various contents (in the case where you are editing things on the worksheet) will work in the same way. i am going to checkout this branch and test this.

**Comment by percyliang** [comment]
It should be fine because things are now getting passed properly down using
props instead of using state.  But go ahead and test it anyway (remember to
get the latest version since I made a few updates).

**Comment by pbhatnagar3** [comment]
yup yup. Everything works without any console errors. This is soooo good :+1:

**Pull request merged**